phase08/StudentGrades/StudentGrades/ProgramRunner.cs

```diff
@@ -0,0 +1,27 @@
+using DefaultNamespace;
+
+namespace project;
+
+public class ProgramRunner
+{
+    public void Run()
+    {
+        var fileReader = new FileReader();
+        var output = new OutputPrinter();
+        var students = new List<Student>();
+        var lessons = new List<StudentGrade>();
+        var dbController = DBController.Instance;
+        var jsonParser = new JsonParser();
+
+        var studentsData = fileReader.ReadFileFromDisk(@"studentsData.json");
+        var scoresData = fileReader.ReadFileFromDisk(@"scoresData.json");
+
+        students = jsonParser.ParseStudents(studentsData);
+        lessons = jsonParser.ParseScores(scoresData);
+
+        dbController.AddRecordsToDB(students, lessons);
+        dbController.calculateAverageGrade();
```

Comment:
here we are using **Postgresql** as a database, imagine that we want to change our database in the future. what we should do? (**O/C principle**)
**+** also, calculating averageGrade is not duty of database.

Reply:
There was a rule that said to develop a program, you should fulfill the required and wanted features and not think of future needings. it might seem to be in conflict with the *O/C principle*. I don't know the border between the two concepts :D.